File: packages/julia/tests_live/test_candidate_registry.jl
using Datamonger
using JSON3

repository_root = normpath(joinpath(@__DIR__, "..", "..", ".."))
release_root = joinpath(repository_root, "registry", "releases", "candidate-0002")
selector_document = JSON3.read(
    read(joinpath(release_root, "selector.json")),
    Dict{String,Any},
)
index = JSON3.read(read(joinpath(release_root, "index.json")), Dict{String,Any})
selected = Datamonger.Registry(
    selector_document["release"],
    selector_document["index_sha256"],
    selector_document["index_url"],
    selector_document["schema_version"],
)
cache_dir = mktempdir(; prefix="datamonger-candidate-0002-")

for dataset in index["datasets"]
    result = fetch_data(
        dataset["name"];
        source=dataset["source"],
        version=dataset["version"],
        registry=selected,
        cache_dir,
        return_info=true,
    )
    records = dataset["representation"]["expect"]["verification"]
    expected = last(records)["digest"]
    result.info.canonical_digest == expected ||
        error("canonical digest mismatch for $(result.info.dataset_id)")
    println(result.info.dataset_id, " ", result.info.canonical_digest)
end

File: packages/r/tests_live/test_candidate_registry.R
pkgload::load_all(".", quiet = TRUE)

args <- commandArgs(trailingOnly = TRUE)
release <- if (length(args)) args[[1L]] else "candidate-0002"
release_root <- file.path("../../registry/releases", release)
selector_document <- jsonlite::fromJSON(
  file.path(release_root, "selector.json"),
  simplifyVector = FALSE
)
index <- jsonlite::fromJSON(
  file.path(release_root, "index.json"),
  simplifyVector = FALSE
)
selected <- do.call(registry_selector, selector_document)
cache_dir <- tempfile(paste0("datamonger-", release, "-"))

for (dataset in index$datasets) {
  result <- fetch_data(
    dataset$name,
    source = dataset$source,
    version = dataset$version,
    registry = selected,
    cache_dir = cache_dir,
    return_info = TRUE
  )
  records <- dataset$representation$expect$verification
  expected <- tail(records, 1L)[[1]]$digest
  stopifnot(identical(result$info$canonical_digest, expected))
  message(result$info$dataset_id, " ", result$info$canonical_digest)
}
